lib/wasix/src/bin_factory/exec.rs

```diff
@@ -38,7 +40,46 @@ pub async fn spawn_exec(
     // any longer
     drop(binary);
 
-    spawn_exec_module(module, env, runtime)
+    struct State {
+        result: Option<Result<TaskJoinHandle, SpawnError>>,
+        waker: Option<Waker>,
+    }
+    let state = Arc::new(Mutex::new(State {
+        result: None,
+        waker: None,
+    }));
+
+    let cloned_state = state.clone();
```

Comment:
I don't quite get what's going on here.

We already have VirtualTaskManagerExt::spawn_await => https://github.com/wasmerio/wasmer/blob/56b3de2dde3119a112c42052645e3329c7c79928/lib/wasix/src/runtime/task_manager/mod.rs#L455